routines/Extended/Speed.asm

```diff
@@ -1,24 +1,34 @@
 
 ;Input:  A   = type of speed update
-;              0 ... x+y with gravity
+;              0 ... x+y with high gravity (YSpeed +3 every flame)
 ;              1 ... x+y without gravity
 ;              2 ... x only
 ;              3 ... y only
+;              4 ... x+y with low gravity (YSpeed +1 every flame)(Lava Lotus's fiery objects)
+;              5 ... x+y with normal gravity (YSpeed +2 every flame)(Hammer,Coin from coin cloud game,etc...)
+;              6 ... x+y with high gravity (YSpeed +3 every flame)(PIXI Orifinal (Same as 0) )
+;              7 ... x+y with very high gravity (YSpeed +4 every flame)(Player fireball)
 
 ?main:
+   CMP #$04 : BCS ?+ : CMP #$00
    BEQ ?.SpriteSpd
    DEC : BEQ ?.SpriteSpdNoGravity
    DEC : BEQ ?.SpriteXSpd
-   DEC : BEQ ?.SpriteYSpd
-   RTL
+   BRA ?.SpriteYSpd
+?+ SBC #$03
+   LDY !extended_y_speed,x
+   BMI $04
```

Comment:
Why is this `BMI $04`, don't use raw values like this please, use labels as appropriate?